### DISCUSSION TITLE
What is Renovate bot github.actor value?

### DISCUSSION THREAD

**Discussion by back-2-95** [renovatebot/renovate#13704] [Request Help] [answered]
For Dependabot created PRs, I can use

```
if: ${{ github.actor == 'dependabot[bot]' }}
```

So what is the value for Renovate bot?

**Comment by rarkins** [comment]
I'm not familiar with that field. Assuming it's not a proprietary thing which GitHub added for Dependabot, Renovate is presumably `renovate[bot]`

**Reply by back-2-95** [reply]
Ah, so the actor name is somehow "generated" by Github?

**Reply by rarkins** [reply]
I've never seen it before

**Reply by back-2-95** [reply]
Ok, I need to debug what Renovate bot gives for ${{ github.actor }}

**Reply by back-2-95** [reply]
I still can match both Dependabot & Renovate with (example on pull requests):

```
if: startsWith(github.event.pull_request.head.ref, 'dependabot/') || startsWith(github.event.pull_request.head.ref, 'renovate/')
```

**Reply by rarkins** [reply]
Is your purpose to simply identify who created it, or for security? Because using the branch prefix like above won't be secure

**Reply by back-2-95** [reply]
Could you elaborate? Also I tested what value the ${{ github.actor }} is. It is like you said: `renovate[bot]`.

We just want to do extra steps on Github Workflow when PR is made by a bot.

**Reply by rarkins** [reply]
You may want to prevent people from impersonating Renovate PRs if you are assigning any extra privilege to them. But it sounds like that's not the case

**Reply by HonkingGoose** [reply]
> I'm not familiar with that field. Assuming it's not a proprietary thing which GitHub added for Dependabot, Renovate is presumably `renovate[bot]`

I found the relevant GitHub Docs that explain the `github` context. 😉 [^gh-docs]

> The `github` context contains information about the workflow run and the event that triggered the run. You can also read most of the `github` context data in environment variables. For more information about environment variables, see "[Using environment variables](https://docs.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables)."

> | Property name | Type | Description |
> | --- | --- | --- |
> | `github.actor` | `string` | The username of the user that initiated the workflow run. |

[^gh-docs]: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

**Comment by mericozkayagan** [comment, marked as answer]
`if: ${{ github.actor == 'renovate[bot]' }}`
this one worked for me
